### DISCUSSION TITLE
A way to block all websites ending with .top

### DISCUSSION THREAD

**Discussion by LucianAndries** [uBlockOrigin/uBlock-issues#1906] [Q&A] [answered]
Hello!

I really dunno where to post this, I'm dumb...

My question is as the title, is there a way to block websites that end in .top and not only? Or should I say the .top domain with all its websites?
Everything from this domain is inserting malicious codes into the websites I follow, and I want to block it. Well, there are some exceptions with .top that I want to allow...

So.......is this "feature" here, or does it need implementing?

I tried "||.top" or "||*.top", but it does not work..... Is there a similar code I can use?

Thank you and happy hollydays!

**Comment by uBlock-user** [comment, marked as answer]
`||top^`

**Reply by LucianAndries** [reply]
And will this block ALL websites ending with .top? .top as in .com, .net, etc...

**Reply by uBlock-user** [reply]
No, only .top domains.

**Reply by LucianAndries** [reply]
"No, only .top domains".............. But that's exactly what I said! Block all websites on the .top domain. There is only 1 domain, the websites registered on it are subdomains.

I tried that code. But it only works half way... It blocks you from going to .top websites. But it does not block .top websites from loading and injecting crap into other NON .top websites...

For example, try the code yourself, then go to this webpage. You will see .top websites not marked with red.
https://aniwatcher.com/shinka-no-mi-shiranai-uchi-ni-kachigumi-jinsei-episode-12

**Reply by uBlock-user** [reply]
> .top as in .com, .net, etc...

^ My response was to that, also that statement makes no sense either.

> https://aniwatcher.com/shinka-no-mi-shiranai-uchi-ni-kachigumi-jinsei-episode-12

Logger shows no .top website appearing.

**Reply by LucianAndries** [reply]
You're using Logger? Just click on uBlock and see the websites...
Also they appear on Logger for me. Try reloading the page until .top shows up.

**Reply by uBlock-user** [reply]
> Also they appear on Logger for me. Try reloading the page until .top shows up.

Then investigate which filter is causing `||top^` to not work as expected.

**Reply by LucianAndries** [reply]
Under the video, try other sources. A .top will show up.

**Comment by LucianAndries** [comment]
Here are some screenshots.

Hmm, making the screenshots showed me the issue, I think... But it should not matter imo.
I fully allowed the video host websites, and the .top crap comes from them. BUT, I blocked the .top domain entirely, so it should not matter despite allowing the video host website........ Is it a bug?

I have to fully allow the video host websites, coz the built-in filters marks those domains as dangerous and blocks them. The website I gave you doesn't host the videos itself, it just uses embeding. So.........yeah.
Or do I need to set it like this for every video host site, to be more efficient? ||top^$domain=xxx
.

![screenshot 1](https://user-images.githubusercontent.com/59364663/147449591-3bbfa8d7-7461-4040-9d53-b2b8b2494883.jpg)
![screenshot 2](https://user-images.githubusercontent.com/59364663/147449596-527d28d5-b2ce-40df-872b-e27166b15a35.jpg)

**Reply by LucianAndries** [reply]
I appreciate all the help btw!!! 👍👍👍

**Reply by uBlock-user** [reply]
You added dynamic `allow` rule for `streamtape.com`, `strtpe.link` and `tapecontent.net` which is why you're seeing `.top` network request happening, terrible idea. Remove all `allow` rules in my Rules pane, they're overiding built-in uBO fllters.

**Reply by LucianAndries** [reply]
Yes, but as I said, removing the "dynamic allow" blocks all the videos! Videos are not loading anymore, because uBO's dumb built-in filters are blocking the video domains.
Kinda what I needed with the .top domain... Friggin irony!

Here, look:
.
![Untitled](https://user-images.githubusercontent.com/59364663/147457685-e3c7d2c6-1d06-4572-af3c-add73bb286ef.jpg)

**Reply by uBlock-user** [reply]
> Yes, but as I said, removing the "dynamic allow" blocks all the videos! Videos are not loading anymore, because uBO's dumb built-in filters are blocking the video domains.

Reset uBO to default settings. The issue you present  is not happening with default settings/lists. Also disable all other extensions you have enabled when testing again.

**Reply by gorhill** [reply]
You have an exception filter `@@*$stylesheet` overriding your `||top^` filter - it's right there in the logger screenshot. Use `||top^$important` if you want to avoid exception filter override.

**Reply by LucianAndries** [reply]
Thank you very much @gorhill, I added it.

Happy Everything to you all!

**Reply by LucianAndries** [reply]
Btw @gorhill, what does "except when / even if" do? How do I use this block option correctly?

Thanks again!

**Reply by gorhill** [reply]
You really should not customize with filters/rules you do not filly understand, you will keep having issues with sites. There is plenty of documentation for filters/rules, see the wiki.

**Reply by uBlock-user** [reply]
He's blocking sites via DF which I'm guessing broke CSS, so to fix that he added `@@*$stylesheet`..., so it would be better if he stays off using DF in the first place.

**Reply by LucianAndries** [reply]
What is DF?
Also nope, I did not add any "@@*$stylesheet" filter. I don't even know what that is and does, lol. That was Green because I Fully Allowed the video host website to not be blocked by the built-in uBO filters. But despite blocking the .top domain, some stuff still managed to slip through...

But as a reminder, everything is fully working now. I even added the "$important" to it.
So, this thread is pretty much solved, imo. Even selected the answer... How do I close it now?

Thank you both for your help!

**Reply by gorhill** [reply]
> I did not add any "@@*$stylesheet" filter

It's there in your logger screenshot, that filter is not from the default filter lists, so you added it somehow. Just use uBO as an install-and-forget blocker, customizing like you did is not for you, you are and will misconfigure uBO and this will break sites or this will seriously lower your protection, that is the reason you were having issues.

**Reply by LucianAndries** [reply]
As I said, I fully allowed the website "stripe.link". It is a video host website.

Now, I can see myself and understand what you are saying. But I don't have "@@*$stylesheet" in My Filters as Allowed.
The ONLY "$stylesheet" in My Filters is this, and it's a Block, not an Allow:
||c.disquscdn.com/next/embed/styles/recommendations.dc416061518569e8bbf0760f47e06ac5.css$stylesheet,important

I dunno why Logger reports that specific thing as Allowed. There must be an error somewhere.

**Reply by uBlock-user** [reply]
> I dunno why Logger reports that specific thing as Allowed. There must be an error somewhere.

It's coming from a filterlist you subscribed. You didn't reset uBO as I instructed so it's there.

**Reply by ghost** [reply]
It can be written as `@@$css`.

**Reply by gorhill** [reply]
>  dunno why Logger reports that specific thing as Allowed

Just click on the filter in the logger, this will tell you where it comes from.

**Comment by LucianAndries** [comment]
Hmm, seems to be working now. All new .top websites are marked as red.
Dunno what I did........ Well, I did this:
- I cleared dynamic allow for the video host websites.
- I added back dynamic allow to them.
- Then I also added "||top^$domain=" for each of them, with "$important".

Dunno which one made it work, but I'm glad it works. Thank you for the help!

**Comment by gorhill** [comment]
> Dunno which made it work

All work fine with default settings/lists.

**Comment by ghost** [comment]
Victim of this thread perhaps https://malwaretips.com/threads/ublock-i-exfiltrate-exploiting-ad-blockers-with-css.111338/page-2#post-967647 ?
